kore/src/Kore/JsonRpc.hs

```diff
@@ -120,9 +130,9 @@ instance FromRequest (API 'Req) where
     parseParams _ = Nothing
 
 data ExecuteState = ExecuteState
-    { state :: !KoreJson
-    , depth :: !Depth
-    , condition :: !(Maybe Condition)
+    { state :: KoreJson
+    , substitution :: Maybe KoreJson
```

Comment:
See prior comment, `substitution`'s type has to change

Reply:
Not necessarily, we can represent substitutions in K like: (X #Equals a) #And (Y #Equals b) ...